pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -156,6 +164,7 @@ func (s *FlinkStateMachine) handleNewOrUpdating(ctx context.Context, application
 	// Create the Flink cluster
 	err := s.flinkController.CreateCluster(ctx, application)
 	if err != nil {
+		s.getAndUpdateError(ctx, application, err)
```

Comment:
This pattern seems a bit error-prone to me, as it requires us to remember to include this before every return err in order to avoid potentially getting stuck (for example, these are missing from several places in handleApplicationSavepointing).

Can we do for all errors in one place?

Reply:
Good point. This has been addressed in the `Handle()` loop. Let me know if that doesn't make sense.